### ISSUE TITLE
manual-combined.xml fails to validate

### ISSUE THREAD

**Issue by Mesabloo** [nix-community/home-manager#1512]
### Issue description

Recently switched to the last commit in this repository (9b1b55ba0264a55add4b7b4e022bdc2832b531f6), when typing `home-manager switch` in my terminal, the new generation cannot be built because of this error:
```
/nix/store/ypag5j9a8gf1flv0za65jga2v28xzk8f-manual-combined/manual-combined.xml:11713: element xref: validity error : IDREF attribute linkend references an unknown ID "opt-home.file._name__.source"
 11709  <simpara>This release has the following notable changes:</simpara>
 11710  <itemizedlist>
 11711  <listitem>
 11712  <simpara>
 11713  The <xref linkend="opt-home.file._name__.source"/> option now allows source files to be
 11714  hidden, that is, having a name starting with the <literal>.</literal> character. It
 11715  also allows the source file name to contain characters not typically

/nix/store/ypag5j9a8gf1flv0za65jga2v28xzk8f-manual-combined/manual-combined.xml fails to validate
builder for '/nix/store/l5af1qhydy6v11pnasmyw0703r6v3h4w-manual-combined.drv' failed with exit code 3
cannot build derivation '/nix/store/mhqpqw81n2v23i1pdvmc0884ys1hckln-man-pages.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/w8j95bpjlz49fn95abhk9azg61g5s4xn-home-manager-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/0nwr91h5639wv480q50i6h2nj3chqigj-home-manager-generation.drv': 1 dependencies couldn't be built
error: build of '/nix/store/0nwr91h5639wv480q50i6h2nj3chqigj-home-manager-generation.drv' failed
```
I am not sure if this is the right place to report the issue. This is a bit weird because the CI seems to pass on the latest commit.

#### Maintainer CC

@rycee I believe

#### Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.64, NixOS, 20.09alpha290.3b8ddb2f1ee (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(root): `"home-manager, nixos-20.09alpha290.3b8ddb2f1ee, nixos-unstable-21.03pre243968.2a35f664394"`
 - channels(mesabloo): `"home-manager, nixos-unstable-21.03pre243968.2a35f664394"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

My home-manager configuration: https://github.com/mesabloo/nix-config

**Comment by cole-h** [comment]
See also: https://github.com/nix-community/home-manager/issues/1479.

**Comment by cole-h** [comment]
I think the problem comes from the fact that you have `programs.home-manager.path` defined, but also have `home-manager` channels. Try removing those `home-manager` channels and check back.

**Comment by Mesabloo** [comment]
Oh I didn't see that this issue was actually already reported. I will try and report any success/failure.

**Comment by Mesabloo** [comment]
Alright, so I did 
```bash
$ nix-channel --remove home-manager
$ sudo nix-channel --remove home-manager
# both just in case
$ nix-channel --update
$ sudo nix-channel --update
# same, just in case
```
Now `nix-channel --list` reports that the `home-manager` channel is not there anymore (which is the expected behavior of the above commands, so that's fine).

I took a look at #1479, tried some commands like `nix run nixpkgs.nix-info --command nix-info ; nix eval '(<home-manager>)'` which returns 
```
system: "x86_64-linux", multi-user?: yes, version: nix-env (Nix) 2.3.7, channels(root): "nixos-20.09alpha290.3b8ddb2f1ee, nixos-unstable-21.03pre243968.2a35f664394", channels(mesabloo): "nixos-unstable-21.03pre243968.2a35f664394", nixpkgs: /nix/var/nix/profiles/per-user/root/channels/nixos
error: file 'home-manager' was not found in the Nix search path (add it using $NIX_PATH or -I), at (string):1:2
```
so I believe that this is normal because the channel does not exist anymore.
However, `home-manager switch` still fails with the same error on `manual-combined`.

I might try setting `manual.manpages.enable = false;` if it doesn't really cause any issue.

**Comment by cole-h** [comment]
You should try the `nix-shell` command I posted in that issue -- for you, it'd be `nix-shell https://github.com/nix-community/home-manager/archive/9b1b55ba0264a55add4b7b4e022bdc2832b531f6.tar.gz -A install`.

Things should start working from there.

**Comment by Mesabloo** [comment]
I'm afraid this will clash with my system-level installation (in `/etc/nixos/configuration.nix`), because that's where I installed home-manager, with this:
```nix
{ config, pkgs, ... }:

{
  imports =
    [ # Include the results of the hardware scan.
      ./hardware-configuration.nix
      "${builtins.fetchTarball https://github.com/rycee/home-manager/archive/9b1b55ba0264a55add4b7b4e022bdc2832b531f6.tar.gz}/nixos"
    ];

# ...

    environment.systemPackages = with pkgs; [
        wget git home-manager
    ];

# ...
}
```

**Comment by cole-h** [comment]
And you tried `nixos-rebuild switch`? If you're managing home-manager as a NixOS module, I don't think `home-manager switch` and similar commands work... (Though I've never tried it.)

**Comment by Mesabloo** [comment]
I tried it, yes. As far as I could see, it always had worked for more than a year and a half (when I was on NixOS 19.03). I had no trouble with `home-manager` until today, with the manual.

**Comment by rycee** [comment]
@Mesabloo You are using a recent version of NixOS but either an old version of HM master or a release branch of HM. If you change to use a recent HM master it should work fine.

The home-manager command is not available when using the NixOS module. If you have it installed then you are using both the module and a standalone installation, which is likely to cause problems.

**Comment by Mesabloo** [comment]
The commit I picked is the last one I could pick on master. I was using the 20.03 release before (I updated all the fetches with the correct commit hash).

That's somehow weird, because I never installed it using the `nix-shell` (just imported the `/nixos` and added `home-manager` to my system packages), and it basically works.

**Comment by Mesabloo** [comment]
Alright. Turns out something was not running well.

I rebooted my machine (I was going to bed, but received the email from your response), updated the path of the fetch in my system configuration (because it still referenced `rycee/home-manager`, so I replaced it with `nix-community/home-manager`), rebuilt the system configuration (no fetch was done at that moment). I tried switching generation again, and now it works (after `home-manager` "reinstalling" itself). I think that the switch did not fully work, and that logging out was the solution to updating the path to the home-manager executable/removing the channels/something else. No more error when building the manual.
